editor/editor_interface.cpp

```diff
@@ -263,6 +265,87 @@ void EditorInterface::set_current_feature_profile(const String &p_profile_name)
 	EditorFeatureProfileManager::get_singleton()->set_current_profile(p_profile_name, true);
 }
 
+// Editor dialogs.
+
+void EditorInterface::popup_dialog_scene_tree(const Callable &p_callback, const TypedArray<StringName> &p_valid_types) {
+	// TODO: Should reuse dialog instance instead of creating a fresh one, but need to rework set_valid_types first.
+	if (dialog_scene_tree) {
+		dialog_scene_tree->queue_free();
+	}
+	dialog_scene_tree = memnew(SceneTreeDialog);
+
+	Vector<StringName> valid_types;
+	int length = p_valid_types.size();
+	valid_types.resize(length);
+	for (int i = 0; i < length; i++) {
+		valid_types.write[i] = p_valid_types[i];
+	}
+	dialog_scene_tree->set_valid_types(valid_types);
+
+	get_base_control()->add_child(dialog_scene_tree);
+
+	dialog_scene_tree->popup_scenetree_dialog();
+
+	const Callable selected_callback = callable_mp(this, &EditorInterface::_node_selected).bind(p_callback);
+	dialog_scene_tree->connect(SNAME("selected"), selected_callback, CONNECT_DEFERRED);
+
+	const Callable canceled_callback = callable_mp(this, &EditorInterface::_node_selection_canceled).bind(p_callback);
+	dialog_scene_tree->connect(SNAME("canceled"), canceled_callback, CONNECT_DEFERRED);
```

Comment:
Extracting callables to a new line like that is pretty unusual. Was this suggested during the review? It's not bad per se, we might want to practice this more.

Reply:
If it was, I've forgotten who suggested it by now xD

I just did it because the `dialog_scene_tree->connect` line was way too long and hard to read otherwise 🙃